packages/syft/src/syft/service/action/action_object.py

```diff
@@ -474,6 +575,40 @@ def make_id(cls, v: Optional[UID]) -> UID:
         """Generate or reuse an UID"""
         return Action.make_id(v)
 
+    class Config:
+        validate_assignment = True
+
+    @pydantic.root_validator()
+    def __check_action_data(cls, values: dict) -> dict:
+        v = values.get("syft_action_data_cache")
+        if values.get("syft_action_data_type", None) is None:
+            values["syft_action_data_type"] = type(v)
+        if not isinstance(v, ActionDataEmpty):
+            if inspect.isclass(v):
+                values["syft_action_data_repr_"] = repr_cls(v)
+            else:
+                values["syft_action_data_repr_"] = (
+                    v._repr_markdown_()
+                    if hasattr(v, "_repr_markdown_")
+                    else v.__repr__()
+                )
+            values["syft_action_data_str_"] = str(v)
+            values["syft_has_bool_attr"] = hasattr(v, "__bool__")
+        return values
+
+    def _save_to_blob_store(self) -> Optional[SyftError]:
+        data = self.syft_action_data
+        if isinstance(data, SyftError):
+            return data
+        if isinstance(data, ActionDataEmpty):
+            print(f"cannot store empty object {self.id}")
+            return data
+        result = self._set_syft_action_data(data)
+        if isinstance(result, SyftError):
+            return result
+        if not TraceResult.is_tracing:
+            self.syft_action_data_cache = self.as_empty_data()
+
```

Comment:
@koenvanderveen

Reply:
TraceResult is used in Plans (check planify method). TraceResult is a global state and when it is enabled, then we're executing a Plan over local variables. At that point we don't want to clear the cache. 

Reason for Setting cache to empty:
This is because, now saving Action Objects include two steps:
- save actual data (in syft_action_data_cache) to blob store
- post that, save the Action Object itself to the Action Store. Now since we don't want to save the actual data present in `syft_action_data_cache` to the ActionStore (as that data is already stored to blob store) we clear it or set it as empty post saving to blob store.